mlrun/api/api/endpoints/functions.py

```diff
@@ -616,14 +614,48 @@ def _get_function_env_var(fn: ServingRuntime, var_name: str):
     return None
 
 
-def _get_project_secret(project_name: str, secret_key: str):
+def _process_model_monitoring_secret(db_session, project_name: str, secret_key: str):
     logger.info(
         "Getting project secret", project_name=project_name, namespace=config.namespace
     )
-    secret_value = mlrun.api.crud.Secrets().get_secret(
-        project_name,
-        mlrun.api.schemas.SecretProviderName.kubernetes,
-        secret_key,
-        allow_secrets_from_k8s=True,
+
+    provider = SecretProviderName.kubernetes
+    secret_value = Secrets().get_secret(
+        project_name, provider, secret_key, allow_secrets_from_k8s=True,
     )
+    user_provided_key = secret_value is not None
+    internal_key_name = Secrets().generate_model_monitoring_secret_key(secret_key)
+
+    if not user_provided_key:
+        secret_value = Secrets().get_secret(
+            project_name,
+            provider,
+            internal_key_name,
+            allow_secrets_from_k8s=True,
+            allow_internal_secrets=True,
+        )
+        if not secret_value:
+            import mlrun.api.utils.singletons.project_member
+
+            project_owner = mlrun.api.utils.singletons.project_member.get_project_member().get_project_owner(
+                db_session, project_name
+            )
+
+            secret_value = project_owner.session
+            if not secret_value:
+                raise MLRunRuntimeError(
+                    f"No model monitoring access key. Failed to generate one for owner of project {project_name}",
+                )
+
+            logger.info(
+                "Filling model monitoring access-key from project owner",
+                project_name=project_name,
+                project_owner=project_owner.username,
+            )
+
+    secrets = SecretsData(provider=provider, secrets={internal_key_name: secret_value})
```

Comment:
There are basically 3 scenarios here:
1. user provided access key
2. access key already exists
3. access key taken from owner
It looks like you log something only for the third one, I guess we should also log something (debug sounds enough) for another one, the less common one, I guess (1)

Reply:
I'll add logs and comments.